classes/Tools.php

```diff
@@ -3351,13 +3351,18 @@ public static function addonsRequest($request, $params = array())
             return false;
         }
 
-        $post_data = http_build_query(array(
+        $post_query_data = array(
             'version' => isset($params['version']) ? $params['version'] : _PS_VERSION_,
             'iso_lang' => Tools::strtolower(isset($params['iso_lang']) ? $params['iso_lang'] : Context::getContext()->language->iso_code),
             'iso_code' => Tools::strtolower(isset($params['iso_country']) ? $params['iso_country'] : Country::getIsoById(Configuration::get('PS_COUNTRY_DEFAULT'))),
             'shop_url' => isset($params['shop_url']) ? $params['shop_url'] : Tools::getShopDomain(),
             'mail' => isset($params['email']) ? $params['email'] : Configuration::get('PS_SHOP_EMAIL')
-        ));
+        );
+        if (isset($params['source']) && ($source=$params['source'])) {
+            $post_query_data['source'] = $source;
```

Comment:
Why not to set $post_query_data as $params['source'] instead of assigning a new variable in the condition?